### ISSUE TITLE
Local variables are propagated to child methods

### ISSUE THREAD

**Issue by gegy1000** [FabricMC/Enigma#105]
Currently when a local variable is remapped within a method, all local variables with the same index in the method hierarchy get the same name applied. Instead, local variables should be unique per method implementation.

**Comment by asiekierka** [comment]
Might be wise to make "arguments" and "local variables" two completely different types of nodes/name types/etc.

**Comment by gegy1000** [comment]
Can be addressed along with #77

**Comment by asiekierka** [comment]
Yes. Just negotiate it with Player so that Matcher can handle the conversion during an eventual update.
